test/unit/black-button_test.js

```diff
@@ -0,0 +1,171 @@
+            idGenerator.generateInternalId = originalGenerateInternalId;
+        });
+
+        it('should update all the device data in the Context Broker entity', utils.checkContextBroker(options));
+
+        it('should return the request id to the device:', function(done) {
+            request(options, function(error, result, body) {
+                should.not.exist(error);
+                result.statusCode.should.equal(200);
+                body.should.equal('#STACK1#0,BT,C,AAAEE1111,,,0$');
+                done();
+            });
+        });
+    });
+
+    describe('When the creation in the CB returns an error: ', function() {
+        it('should return an explanation of the kind of error to the device');
+    });
+
+    describe('When a polling operation arrives from the device: ', function() {
+        var options = {
+                url: 'http://localhost:' + config.thinkingThings.port + config.thinkingThings.root + '/Receive',
+                method: 'POST',
+                form: {
+                    cadena: '#STACK1#5,BT,P,AAAEE1111,0$'
+                }
+            },
+            originalGenerateInternalId;
+
+        beforeEach(function(done) {
+            config.ngsi.plainFormat = true;
+
+            originalGenerateInternalId = idGenerator.generateInternalId;
+            idGenerator.generateInternalId = mockedGenerateInternalId;
+
+            async.series([
+                utils.prepareMocks(
+                    './test/unit/contextRequests/blackButtonPollingRequestQuery.json',
+                    './test/unit/contextResponses/blackButtonPollingRequestQuerySuccess.json',
+                    '/v1/queryContext'),
+                utils.prepareMocks(
+                    './test/unit/contextRequests/blackButtonPollingRequestUpdate.json',
+                    './test/unit/contextResponses/blackButtonPollingRequestUpdateSuccess.json',
+                    '/v1/updateContext')
+            ], done);
+        });
+
+        afterEach(function() {
+            config.ngsi.plainFormat = false;
+            idGenerator.generateInternalId = originalGenerateInternalId;
+        });
+
+        it('should query the Context Broker for the current state', utils.checkContextBroker(options));
+        it('should return the extra data and state if available', function(done) {
+            request(options, function(error, result, body) {
+                should.not.exist(error);
+                result.statusCode.should.equal(200);
+                body.should.equal('#STACK1#5,BT,P,AAAEE1111:C.S:999999,0$');
+                done();
+            });
+        });
+    });
+    describe('When a polling operation arrives from the device and the request was failed: ', function() {
+        it('should return the appropriate error code');
+        it('should return the extra information if available');
+    });
+    describe('When a request close operation arrives from the device: #STACK1#1,BT,X,86,0$', function() {
```

Comment:
At some point you will add a blank line just before this describe() for clarity... :p